### PR TITLE
Scaffolding for CI environment setup in Azure

### PR DESCRIPTION
**Pull request by jhoreman**
### Which issue this PR addresses:
https://issues.redhat.com/browse/ARO-10133

### What this PR does / why we need it:

Scaffolding for CI env buildouts as per our go generator/ARM deploy scripts. Mostly a stub for now as it only lays down a resource group and an ACR. Meant to be added to when we deploy more infra in the CI/E2E sub.

### Test plan for issue:

<!--
How did you test that this PR works?

Local testing in RH Test sub. 

### Is there any documentation that needs to be updated for this PR?

Included in PR

### How do you know this will function as expected in production? 

Not a production feature.

### CODE REVIEW

File: pkg/deploy/generator/templates_ci.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+		"acrResourceId",
+	}
+
+	for _, param := range params {
+		p := &arm.TemplateParameter{Type: "string"}
+		switch param {
+		case "acrLocationOverride":
+			p.DefaultValue = "eastus"
+		case "acrName":
+			p.DefaultValue = "arosvcdev"
```
Comment (SudoBrendan) [inline pkg/deploy/generator/templates_ci.go:25]: I think our MVP ACR was `arodevsvc`? @shubhadapaithankar can you confirm (my portal auth is acting up again)

**Review by SudoBrendan — APPROVED** [review]
LGTM - if we can let's validate the default ACR Name is correct

File: pkg/deploy/generator/templates_ci.go (hunk 1 above)
Comment (shubhadapaithankar) [reply pkg/deploy/generator/templates_ci.go:25]: yes , I am using `arosvcdev` ACR  from CI tenant

Comment (jhoreman) [reply pkg/deploy/generator/templates_ci.go:25]: Thanks for confirming Shubhada!

**Review by shubhadapaithankar — APPROVED** [review]
LGTM

**Pull request merged**